pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -81,11 +81,9 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 
 	clusters = append(clusters, configgen.buildOutboundClusters(env, proxy, push)...)
 
-	// compute the proxy's locality. See if we have a CDS cache for that locality.
-	// If not, compute one.
-	locality := proxy.Locality
-	if locality != nil {
-		applyLocalityLBSetting(locality, clusters, env.Mesh.LocalityLbSetting, false)
+	// apply load balancer setting fot cluster endpoints
+	if pilot.EnableLocalityLoadBalancing() && proxy.Locality != nil {
```

Comment:
As i added this check, should be after  #15014

Reply:
the rest is fine but I don't think we want to change this behavior now (without 15014)

Reply:
I think this should be added when the feature was introduced, it is a bug.